t5/t5.py

```diff
@@ -5,8 +5,8 @@
 import mlx.core as mx
 import mlx.nn as nn
 import numpy as np
-from mlx.utils import tree_map, tree_unflatten
-from transformers import T5Config, T5Tokenizer
+from mlx.utils import tree_map, tree_flatten, tree_unflatten
```

Comment:
Why include `tree_flatten`?

Reply:
Ooops. My bad. I added it to confirm all weights were initialized and that the shapes matched. Perhaps that's something we could implement in `mlx.utils`?

Reply:
Yea I've been thinking about something like that. I've had annoying bugs with that in the past. Actually I was wondering if it made sense to have `Module.update` have a mode where it warns or throws if:

1. Not all the parameters were updated
2. If the shapes of updated parameters are unexpected.

Reply:
Nice, that would be great! I'm using this to diagnose any issues:

```python
    current_weights = tree_flatten(model.parameters())
    weights_to_load = list(weights.items())
    current_weights_dict = dict(current_weights)
    current_weights_keys = set(current_weights_dict.keys())
    weights_to_load_dict = dict(weights_to_load)
    weights_to_load_keys = set(weights_to_load_dict.keys())
    print("Missing weights: ", sorted(current_weights_keys - weights_to_load_keys))
    print()
    print("Weights ignored: ", sorted(weights_to_load_keys - current_weights_keys))
    for key in current_weights_keys & weights_to_load_keys:
        if weights_to_load_dict[key].shape != current_weights_dict[key].shape:
            print("Shape mismatch for key: ", key)
            print("Expected shape: ", current_weights_dict[key].shape)
            print("Loading shape: ", weights_to_load_dict[key].shape)
```